stdlib/public/core/SmallString.swift

```diff
@@ -241,22 +241,31 @@ extension _SmallString {
   fileprivate mutating func withMutableCapacity(
     _ f: (UnsafeMutableRawBufferPointer) throws -> Int
   ) rethrows {
-    let len = try withUnsafeMutableBytes(of: &self._storage) {
-      (rawBufPtr: UnsafeMutableRawBufferPointer) -> Int in
-      let len = try f(rawBufPtr)
-      UnsafeMutableRawBufferPointer(
-        rebasing: rawBufPtr[len...]
-      ).initializeMemory(as: UInt8.self, repeating: 0)
-      return len
-    }
-    if len == 0 {
+    let len = try withUnsafeMutableBytes(of: &_storage, f)
+
+    if len <= 0 {
       self = _SmallString()
       return
     }
-    _internalInvariant(len <= _SmallString.capacity)
+    zeroTrailingBytes(of: &_storage, from: len)
+    self = _SmallString(leading: _storage.0, trailing: _storage.1, count: len)
+  }
+}
 
-    let (leading, trailing) = self.zeroTerminatedRawCodeUnits
-    self = _SmallString(leading: leading, trailing: trailing, count: len)
+@inline(__always)
+private func zeroTrailingBytes(
+  of storage: inout _SmallString.RawBitPattern, from len: Int
+) {
+  _internalInvariant(len > 0)
+  _internalInvariant(len <= _SmallString.capacity)
+  if len <= 0 {
+    storage.0 = 0
+    storage.1 = 0
+  } else if len <= 8 {
+    storage.0 &= ((~0) &>> (8 &* (8 &- len)))
+    storage.1 = 0
+  } else {
+    storage.1 &= ((~0) &>> (8 &* (16 &- len)))
```

Comment:
What happens to `storage.0`? Could you repro the un-cleared bytes in a unit test?

Reply:
What uncleared bytes?

Reply:
Nevermind